WcaOnRails/app/models/post.rb

```diff
@@ -26,7 +26,7 @@ def edit_path
     if is_crash_course_post?
       Rails.application.routes.url_helpers.delegate_crash_course_edit_path
     else
-      Rails.application.routes.url_helpers.edit_post_path(slug)
+      Rails.application.routes.url_helpers.edit_post_path(id: slug)
```

Comment:
This was a necessary change, and I think it's a bug in Rails.
I think I ran into this one : https://github.com/rails/rails/issues/12178.

I got similar weird outputs from byebug : 
```
[27, 36] in /vagrant/WcaOnRails/app/models/post.rb
   27:       Rails.application.routes.url_helpers.delegate_crash_course_edit_path
   28:     else
   29:       Rails.application.routes.url_helpers.edit_post_path(id: slug)
   30:     end
   31:     byebug
=> 32:   end
   33: 
   34:   def update_path
   35:     if is_crash_course_post?
   36:       Rails.application.routes.url_helpers.delegate_crash_course_path
(byebug) Rails.application.routes.url_helpers.edit_post_path(id: slug)
"/fr/posts/we-need-to-generate-the-open-source-ai-alarm/edit"
(byebug) Rails.application.routes.url_helpers.edit_post_path(slug)
*** ActionController::UrlGenerationError Exception: No route matches {:action=>"edit", :controller=>"posts", :id=>nil, :locale=>"we-need-to-generate-the-open-source-ai-alarm"} missing required keys: [:id]

nil
(byebug) Rails.application.routes.url_helpers.edit_post_path(titi:'a')
*** ActionController::UrlGenerationError Exception: No route matches {:action=>"edit", :controller=>"posts", :locale=>:fr, :titi=>"a"} missing required keys: [:id]
```

As you can see the `default_url_options` are correctly taken into account as soon as you specify an additional option hash (the last route generation is expected to fail, but you can see that locale is correctly detected !).
According to the related issue, Rails 4 uses an optimized url generation which doesn't use properly the `default_url_options`, setting any option hash bypasses this optimized generation.

Hopefully there was only two uses of the `url_helpers` with params.